### PR TITLE
References between steps

### PR DESCRIPTION
**Pull request by philtweir**
### What does this change do?

* Adds the ability for steps to reference each other.
* Changes Workflow passing through `run` construction from singleton to fold

### How to review?

Make example to see if two or ideally more dask delayeds can be constructed into a graph.

### What is outstanding for this to be merged?

* #1 must be merged
* docs and examples

### CODE REVIEW

File: .github/workflows/_build.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from portray import config, render
```
Comment (philtweir) [inline .github/workflows/_build.py:1]: This is CI build code - it can be tidied up, but the medium term solution is to improve portray.

Comment (philtweir) [reply .github/workflows/_build.py:1]: Short explanation: portray wants the configuration (including for mkdocs) in TOML. mkdocs extensions use YAML features for their config that TOML doesn't have. Therefore, we need to hijack the config loading.

File: tests/test_cwl.py [hunk 2]
```diff
@@ -11,6 +11,21 @@ def increment(num: int) -> int:
...
     return num + 1
 
+@task()
+def double(num: int) -> int:
+    """Double an integer."""
+    return 2 * num
+
+@task()
+def mod10(num: int) -> int:
+    """Double an integer."""
```
Comment (elleryames) [inline tests/test_cwl.py:21]: Copy and paste typo.

**Comment by philtweir** [comment]
Following discussion in-person and to reduce duplication of effort, merging these into one pre-release PR.

**Pull request closed without merging**